src/core/common/message.cpp

```diff
@@ -944,14 +940,13 @@ Message *PriorityQueue::GetTail(void) const
     return FindFirstNonNullTail(0);
 }
 
-otError PriorityQueue::Enqueue(Message &aMessage)
+void PriorityQueue::Enqueue(Message &aMessage)
 {
-    otError  error = OT_ERROR_NONE;
     uint8_t  priority;
     Message *tail;
     Message *next;
 
-    VerifyOrExit(!aMessage.IsInAQueue(), error = OT_ERROR_ALREADY);
+    VerifyOrExit(!aMessage.IsInAQueue(), OT_NOOP);
```

Comment:
```suggestion
    VerifyOrExit(aMessage.GetPriorityQueue() != this, OT_NOOP);

    OT_ASSERT(!aMessage.IsInAQueue());
```
Just let the cases that the message is no another queue to fail.